### PR TITLE
Switch to using Requests instead of urllib2

### PR DESCRIPTION
**Pull request by rossgray**

### CODE REVIEW

File: pyticketswitch/interface.py [hunk 1]
```diff
@@ -44,9 +44,13 @@ def __init__(
 
         if not additional_elements:
             additional_elements = {}
-
         self.additional_elements = additional_elements
 
+        # If no Requests session create a new one
+        if not requests_session:
```
Comment (sholmesy) [inline pyticketswitch/interface.py]: Whats the benefit of allowing a request session to be passed in? I have no idea,  but is there some sort of security issue or anything? idk, just bringing it up.

**Comment by nicwest** [comment]
Looks good to me :D

File: pyticketswitch/interface.py (hunk 1 above)
Comment (nicwest) [inline pyticketswitch/interface.py]: so you can pass it between instances of the API presumably.

File: setup.py [hunk 2]
```diff
@@ -13,6 +13,9 @@
     license='MIT',
     description='A Python interface for the Ingresso XML Core API',
     long_description=open('README.rst').read(),
+    install_requires=[
+        'requests>=2.0.0',
+    ],
     classifiers=[
         'Development Status :: 5 - Production/Stable',
         'Programming Language :: Python :: 2.7',
```
Comment (sholmesy) [inline setup.py]: Seems fine.

Will this break existing users? (obvs they don't have to pull the new build, just curious)

Comment (rossgray) [inline setup.py]: it makes the `requests` library a requirement from now on

**Pull request merged**